localstack-core/localstack/services/stepfunctions/asl/utils/boto_client.py

```diff
@@ -1,19 +1,42 @@
+from typing import Optional
+
 from botocore.client import BaseClient
 from botocore.config import Config
 
 from localstack.aws.connect import connect_to
 from localstack.services.stepfunctions.asl.component.common.timeouts.timeout import TimeoutSeconds
+from localstack.services.stepfunctions.asl.component.state.state_execution.state_task.credentials import (
+    ComputedCredentials,
+    Credentials,
+)
+from localstack.utils.aws.client_types import ServicePrincipal
+
+_BOTO_CLIENT_CONFIG = config = Config(
+    parameter_validation=False,
+    retries={"max_attempts": 0, "total_max_attempts": 1},
+    connect_timeout=TimeoutSeconds.DEFAULT_TIMEOUT_SECONDS,
+    read_timeout=TimeoutSeconds.DEFAULT_TIMEOUT_SECONDS,
```

Comment:
nit: Some learnings from the Lambda PR: you can also set `tcp_keepalive=True` in order to prevent the TCP connection dropping prior to the connect/read timeouts.

From [boto3 docs](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html):
> tcp_keepalive (bool) –
> Enables the TCP Keep-Alive socket option used when creating new connections if set to True.

Reply:
Thanks! I am adding this